### PR TITLE
Adds the Isatoa and Ythralian languages

### PR DESCRIPTION
**Pull request by Draggeru**
As the title says:

Isatoa is the language that originates from Mu, and is available in the language menu

Ythralian is a language designed for characters like Finster, Randall, and any character that's eldritch or otherwise ancient in nature. Speaking the language around beings that don't know it causes brain damage. This language is available via admins only, due to it causing actual damage to people.

If something breaks it's Melberts fault because he bullied me into learning something.

Icons:

icons

### CODE REVIEW

**Comment by Draggeru** [comment]
>Almost all checks fail

This is why I learn nothing new.

File: maplestation_modules/code/modules/language/isatoan.dm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	)
+	spoken_languages = list(
+		/datum/language/common = list(LANGUAGE_ATOM),
+		/datum/language/eldritch = list(LANGUAGE_ATOM),
+	)
+
+/mob/living/translate_language(atom/movable/speaker, datum/language/eldritch, raw_message, list/spans, list/message_mods = list())
+	. = ..()
+	if(. == raw_message)
+	return
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/language/isatoan.dm:61]:
```suggestion
		return
```
needs indentation

File: maplestation_modules/code/modules/language/isatoan.dm [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	if(isnull(language))
+	return
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/language/isatoan.dm:63]:
```suggestion
		return
```

File: maplestation_modules/code/modules/language/isatoan.dm [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	var/datum/language/dialect = GLOB.language_datum_instances[language]
+	dialect.heard_by_mob_who_lacks_critical_information(src)
+
+/datum/language/proc/heard_by_mob_who_lacks_critical_information(mob/living/listener)
+	return
+
+/datum/language/eldritch/heard_by_mob_who_lacks_critical_information(mob/living/listener)
+	listener_real.adjustOrganLoss(ORGAN_SLOT_BRAIN, 10, 100)
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/language/isatoan.dm:71]:
```suggestion
	listener.adjustOrganLoss(ORGAN_SLOT_BRAIN, 10, 100)
```
that said im not sure if adjustOrganLoss works on mob/living, might have to cast to mob/living/carbon/human

File: maplestation_modules/code/modules/language/isatoan.dm [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+		"se", "lu", "xi", "th", "a", "i", "e", "za", "rn", "rk", "vo", "qu", "pi",
+		"iy", "ji", "vu", "gai", "fel", "ceo", "eon", "du", "li", "ios", "ma", "miv", "ola",
+		"lis", "aoi", "rin", "tav", "mu", "vel", "rx", "eri",
+	)
+	icon_state = "mu"
+	icon = 'maplestation_modules/icons/misc/language.dmi'
+	default_priority = 80
+	available_as_pref = TRUE
+
+/datum/language_holder/skrell
```
Comment (CRITAWAKETS) [inline maplestation_modules/code/modules/language/isatoan.dm:20]: Is this meant to be skrell?

**Review by CRITAWAKETS — CHANGES_REQUESTED** [review]
Currently this doesn't compile so uh, might wanna fix that!

**Comment by CRITAWAKETS** [comment]
Eldritch/Ythralian should likely be made to have it's brain-damage effect be resisted via a trait rather than understanding the language, since it seems likely that things like divine entities (or people who are Built Different) wouldn't be affected, but would also not understand the language.

**Comment by Draggeru** [comment]
> Eldritch/Ythralian should likely be made to have it's brain-damage effect be resisted via a trait rather than understanding the language, since it seems likely that things like divine entities (or people who are Built Different) wouldn't be affected, but would also not understand the language.

When you help me code it

File: maplestation_modules/code/modules/language/isatoan.dm (hunk 4 above)
Comment (Draggeru) [reply maplestation_modules/code/modules/language/isatoan.dm:20]: Noh

File: maplestation_modules/code/modules/language/isatoan.dm [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	)
+
+/mob/living/translate_language(atom/movable/speaker, datum/language/language, raw_message, list/spans, list/message_mods = list())
+	. = ..()
+	if(. == raw_message)
+		return
+	if(isnull(language))
+		return
+	var/datum/language/dialect = GLOB.language_datum_instances[language]
+	dialect.heard_by_mob_who_lacks_critical_information(src)
```
Comment (MrMelbert) [inline maplestation_modules/code/modules/language/isatoan.dm:65]:
```suggestion
	dialect.heard_by_mob_who_lacks_critical_information(src, speaker)
```

File: maplestation_modules/code/modules/language/isatoan.dm [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+/datum/language/proc/heard_by_mob_who_lacks_critical_information(mob/living/listener)
```
Comment (MrMelbert) [inline maplestation_modules/code/modules/language/isatoan.dm:67]:
```suggestion
/datum/language/proc/heard_by_mob_who_lacks_critical_information(mob/living/listener, atom/movable/speaker)
```

File: maplestation_modules/code/modules/language/isatoan.dm [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+		"yt", "th", "y", "t", "ra", "li", "an", "ia", "ny", "ar", "la", "in", "ir",
+		"ha", "na", "tr", "al", "yi", "ni", "ay", "ii", "ll", "ry", "xz", "yth",
+		"ran", "lii", "lar", "thr", "ayl", "axl", "nal", "irn", "lah", "nax", "yth", "iar", "uv",
+		"se", "lu", "xi", "th", "a", "i", "e", "za", "rn", "rk", "vo", "qu", "pi",
+		"ith", "h", "r", "n", "fel", "er", "eon", "yl", "ai", "ios", "wa", "va", "mv",
+		"gx", "w", "z", "ei", "eo", "vot", "pr", "us",
+	)
+	icon_state = "eldritch"
+	icon = 'maplestation_modules/icons/misc/language.dmi'
+	default_priority = 120
```
Comment (MrMelbert) [inline maplestation_modules/code/modules/language/isatoan.dm:46]: We talked about the curator thing already but this being 120 priority means I think curators will spawn with this language as their default language

File: maplestation_modules/code/modules/surgery/organs/tongue.dm [hunk 8]
```diff
@@ -6,6 +6,7 @@
 		/datum/language/ratvarian,
 		/datum/language/skrell,
 		/datum/language/yangyu,
+		/datum/language/isatoa,
```
Comment (MrMelbert) [inline maplestation_modules/code/modules/surgery/organs/tongue.dm:9]: Insert this in alphabetical order :)

File: maplestation_modules/code/modules/language/isatoan.dm (hunk 7 above)
Comment (Draggeru) [reply maplestation_modules/code/modules/language/isatoan.dm:46]: I forgor D:

**Review by MrMelbert — APPROVED** [review]
I'm gonna come up with a solution for curators / miners getting it myself if you don't, fair warning

**Pull request merged**